### PR TITLE
Xamarin tutorial part 3

### PR DESCRIPTION
**Pull request by vasarhelyia**

### CODE REVIEW

**Comment by vasarhelyia** [comment]
@bloodrabbit check plz

File: source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: Xamarin CI with Bitrise Tutorial part 3 - Deploy
+date: 2016-03-22 13:54 UTC
+tags: xamarin, tutorial
+authors: Agnes Vasarhelyi|agnes@bitrise.io
+---
+
+There are plenty of ways to deploy your iOS, Android, or Xamarin applications from [Bitrise](https://bitrise.io) to any service of your wish. Let me show you what are your options today, if you would like to deliver your apps to your favorite testers, dearest customers, or the iTunes Connect/Google Play directly.
```
Comment (bloodrabbit) [inline source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown]: what your options are today

File: source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+### Amazon S3
+
+If you'd like to upload your build artifacts to [Amazon S3](https://aws.amazon.com/s3/), you need to add the [`Deploy to Amazon S3`](https://github.com/bitrise-io/steps-amazon-s3-deploy) step to your workflow. After passing your access keys and bucket name to the  step, you can specify a bunch of things, like the access level of the file to upload, its path in your bucket, etc.
+
+## Integrated third-party services
+
+Most of these third-parties has a dedicated iOS, and an Android step on Bitrise. You can use them by simply adding them to your workflow. Check out all the services you can add this easy at our [integrations page](https://www.bitrise.io/integrations).
+
+> Don't forget: you can always add a Script step and install, or run anything you want, but there's no step for it yet. If you feel like contributing, [turn your script into a Bitrise build step](https://github.com/bitrise-io/bitrise-contrib/blob/master/README.md) and earn a 50% discount from our [pro plan with 1 container](https://www.bitrise.io/pricing).
```
Comment (bloodrabbit) [inline source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown]: container -> concurrency

**Comment by viktorbenei** [comment]
👍

File: source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+The first step of deployment to build something to be deployed. You can create archives for your apps with adding the proper steps to your workflow. It's [`Xcode Archive`](https://github.com/bitrise-io/steps-xcode-archive) for iOS, and a simple [`Gradle runner`](https://github.com/bitrise-io/steps-gradle-runner) for Android.
+
+### Xamarin apps
+
+When you're building a Xamarin app on Bitrise, you'll definitely have a [`Xamarin builder`](https://github.com/bitrise-steplib/steps-xamarin-builder) step in your workflow, it's part of the generated one by default. Bitrise detects and parses your configuration file, so it'll set handy environment variables based on it.
+
+![Xamarin Builder step](xamarin_builder_editor.png)
+
+If your configuration has a _real device_ as platform and `Release` as build configuration set, the step will archive your apps after building them automatically. There is no need to add an extra Xcode archive, or Gradle step.
```
Comment (birmacher) [inline source/2016-03-22-xamarin-ci-with-bitrise-tutorial-part-3-deploy.html.markdown]: only real device is required

**Pull request merged**